### PR TITLE
Add Edgecore platform with 7 templates

### PR DESCRIPTION
**Pull request by PavloSkliarenko**
Templates for Edgecore commands:

`show interfaces brief`
`show interfaces status`
`show interfaces switchport`
`show ip interface`
`show system`
`show version`
`show vlan`

### CODE REVIEW

**Comment by PavloSkliarenko** [comment]
@itdependsnetworks @jmcgill298 @jvanderaa Should I make separate PR for each template when it's totally new device platform?

**Comment by jvanderaa** [comment]
Generally we are going for one PR per template update/inclusion for discussion sake. At this point that is water under the bridge and keep this in place.

File: ntc_templates/templates/edgecore_show_interfaces_brief.textfsm [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Value INTERFACE (\S+\s+\d+\s*/\s*\d+)
+Value NAME (.*?)
+Value STATUS (\S+)
+Value PVID (\d+)
+Value PRI (\d+)
```
Comment (jvanderaa) [inline ntc_templates/templates/edgecore_show_interfaces_brief.textfsm:5]: What is `PRI` in this instance? Is it priority or something related to a PRI on the T1 world?

File: ntc_templates/templates/edgecore_show_interfaces_brief.textfsm [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+Value SPEED_DUPLEX (\S+)
```
Comment (jvanderaa) [inline ntc_templates/templates/edgecore_show_interfaces_brief.textfsm:6]: Break up into `SPEED` and `DUPLEX` individually

File: ntc_templates/templates/edgecore_show_interfaces_status.textfsm [hunk 3]
```diff
@@ -0,0 +1,75 @@
+Value Required INTERFACE (.*)
```
Comment (jvanderaa) [inline ntc_templates/templates/edgecore_show_interfaces_status.textfsm:1]:
```suggestion
Value Required PORT (.*)
```

File: ntc_templates/templates/edgecore_show_interfaces_status.textfsm [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+Value PORT_TYPE (.*)
+Value ADDRESS (.*)
+Value NAME (.*)
+Value PORT_ADMIN (.*)
+Value MDIX_MODE (.*)
+Value SPEED_DUPLEX (.*)
```
Comment (jvanderaa) [inline ntc_templates/templates/edgecore_show_interfaces_status.textfsm:7]: Break into multiple

File: ntc_templates/templates/edgecore_show_interfaces_status.textfsm [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+Value VLAN_TRUNKING (.*)
+Value LACP (.*)
+Value PORT_SECURITY (.*)
+Value MAX_MAC_COUNT (.*)
+Value PORT_SECURITY_ACTION (.*)
+Value MEDIA_TYPE (.*)
+Value GIGA_PHY_MODE (.*)
+Value LINK_STATUS (.*)
+Value OPERATION_STATUS (.*)
+Value OPERATION_SPEED_DUPLEX (.*)
```
Comment (jvanderaa) [inline ntc_templates/templates/edgecore_show_interfaces_status.textfsm:25]: Break into multiples

File: ntc_templates/templates/edgecore_show_ip_interface.textfsm [hunk 6]
```diff
@@ -0,0 +1,11 @@
+Value IP (\S+)
+Value NETMASK (\S+)
+Value IFACE (.*)
```
Comment (jvanderaa) [inline ntc_templates/templates/edgecore_show_ip_interface.textfsm:3]:
```suggestion
Value INTERFACE (.*)
```

File: ntc_templates/templates/edgecore_show_system.textfsm [hunk 7]
```diff
@@ -0,0 +1,49 @@
+Value DESCRIPTION (.*)
+Value OID (\d+(\.\d+)*)
+Value UP_TIME (.*)
```
Comment (jvanderaa) [inline ntc_templates/templates/edgecore_show_system.textfsm:3]:
```suggestion
Value UPTIME (.*)
```

File: ntc_templates/templates/edgecore_show_vlan.textfsm [hunk 8]
```diff
@@ -0,0 +1,37 @@
+Value VLAN_ID (\d+)
+Value TYPE (\S+)
+Value NAME (.*)
```
Comment (jvanderaa) [inline ntc_templates/templates/edgecore_show_vlan.textfsm:3]:
```suggestion
Value VLAN_NAME (.*)
```

**Review by jvanderaa — DISMISSED** [review]
Requesting a few changes to align with some of the existing templates, which will allow in a multi-vendor environment to have the transformation of keys to being the same without requiring it in an ETL process.

File: ntc_templates/templates/edgecore_show_interfaces_brief.textfsm (hunk 1 above)
Comment (PavloSkliarenko) [reply ntc_templates/templates/edgecore_show_interfaces_brief.textfsm]: This is "Priority for Untagged Traffic". Changed the name PRI -> PRIORITY

File: ntc_templates/templates/edgecore_show_interfaces_brief.textfsm (hunk 2 above)
Comment (PavloSkliarenko) [reply ntc_templates/templates/edgecore_show_interfaces_brief.textfsm]: Splited it to the 3 new values:
 - negotiation
 - speed
 - duplex

File: ntc_templates/templates/edgecore_show_interfaces_status.textfsm (hunk 4 above)
Comment (PavloSkliarenko) [reply ntc_templates/templates/edgecore_show_interfaces_status.textfsm]: Renamed it in "negotiation" value

File: ntc_templates/templates/edgecore_show_interfaces_status.textfsm (hunk 5 above)
Comment (PavloSkliarenko) [reply ntc_templates/templates/edgecore_show_interfaces_status.textfsm]: Splited it to the 2 new values:
 - speed
 - duplex

**Comment by PavloSkliarenko** [comment]
@jvanderaa Tests are failing because of [tests/cisco_ios/show_ip_interface/cisco_ios_show_ip_interface7.yml](https://github.com/networktocode/ntc-templates/blob/master/tests/cisco_ios/show_ip_interface/cisco_ios_show_ip_interface7.yml). I've made [PR](https://github.com/networktocode/ntc-templates/pull/1558), which fixed it

**Comment by AlexLardschneider** [comment]
Can this be merged?

**Comment by samk-acw** [comment]
I'm keen for this to be merged but I gave it a test and found an issue: I have edgecores with SFP ports, which report the type as '10GBASE SFP+' - the templates don't work with this due to the space. Changing TYPE to `(.*?)` seems to fix it.

**Pull request merged**